bot/seasons/halloween/spookyrating.py

```diff
@@ -0,0 +1,77 @@
+import bisect
+import json
+import logging
+import random
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Colours
+
+log = logging.getLogger(__name__)
+
+with Path('bot', 'resources', 'halloween', 'spooky_rating.json').open() as file:
+    SPOOKY_DATA = json.load(file)
+    SPOOKY_DATA = sorted((int(key), value) for key, value in SPOOKY_DATA.items())
+
+
+class SpookyRating:
+    """
+    A cog for calculating one's spooky rating
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    @commands.cooldown(rate=1, per=5, type=commands.BucketType.user)
+    async def spookyrating(self, ctx, who: discord.Member = None):
+        """
+        Calculates the spooky rating of someone.
+
+        This command accepts an optional user as an argument, defaulting to the command executor.
+        Users are converted from:
+          - User ID
+          - Mention
+          - name#discrim
+          - name
+          - nickname
+
+        Any user will always yield the same result, no matter who calls the command
+        """
+
+        if who is None:
+            who = ctx.author
+
+        # This ensures that the same result over multiple runtimes
+        random.seed(who.id)
```

Comment:
This will change the _global_ seed, which means (since this command gets constants as its input) it will render other commands completely useless. Consider I used this command on myself, then used the terning command 3 times. Well if I used this command again the output of those dice rolls would be the exact same.

This can trivially be fixed by doing
```py
r = random.Random()
r.seed(...)
# use r here
```
You could even make `r` be a member of the class in `__init__` since it'll get seeded each call anyway. (could also come up with a better name?)

I don't know if this is actually a concern for the bot, but since the fix is trivial I decided to comment anyway.